### PR TITLE
Update write.configs.FATES.R

### PR DESCRIPTION
**Pull request by Hhh-hyc**
## Description

To update write.configs file in fates model, the goal is divided into three stages:
1. update machine configuration file to successfully generate template.job
2. update domain, surface and datm files
3. update parameters
## Motivation and Context

The first PR aims to generate basic template.job to run fates model in pecan workflow. However, the web interface of Bety doesn't work on the local machine. Thus, address of local dataset was used.

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [x] When possible

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] My name is in the list of CITATION.cff
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -120,9 +120,7 @@ authors:
   - given-names: Eric R. Scott
     affiliation: University of Arizona
     orcid: 'https://orcid.org/0000-0002-7430-7879'
-  - given-names: Harunobu Ishii
```
Comment (mdietze) [inline CITATION.cff:123]: Unclear why you need to remove someone else's contribution

File: docker-compose.dev.yml [hunk 2]
```diff
@@ -2,6 +2,8 @@
 # 
 # docker-compose -f docker-compose.yml -f docker-compose.dev.yml
 
+version: '3.2'
```
Comment (mdietze) [inline docker-compose.dev.yml:5]: does this PR really need to touch docker files?

File: models/fates/R/write.configs.FATES.R [hunk 3]
```diff
@@ -19,24 +19,40 @@
...
+trait.values$temperate.coniferous <- list()
+trait.values$env <- list()
+
+tc_traits <- as.data.frame(matrix(c(9999, 8888), nrow=1, ncol=2))
+colnames(tc_traits) <- c("leaf_turnover_rate", "veg_respiration_Q10")
+# Fates names for these params are "fates_leaf_long", "q10_mr"
+
+trait.values$temperate.coniferous <- tc_traits
+settings <- '/pecan/models/fates/R/test.fates.xml'
+run.id <- 'test_oneyear'
```
Comment (mdietze) [inline models/fates/R/write.configs.FATES.R:35]: Was this supposed to be an example? You can't have code like this that's outside of function.

File: models/fates/R/write.configs.FATES.R [hunk 4]
```diff
@@ -19,24 +19,40 @@
...
+
+# example trait.values list
+
 write.config.FATES <- function(defaults, trait.values, settings, run.id){
 
    ## site information
    site <- settings$run$site
-   site.id <- as.numeric(site$id)
+   site.id <- site$id # change1: 772 -> SOD1
```
Comment (mdietze) [inline models/fates/R/write.configs.FATES.R:43]: not following the logic of the change here. Comment is also unclear to a general user

File: models/fates/R/write.configs.FATES.R [hunk 5]
```diff
@@ -19,24 +19,40 @@
...
    outdir       <- file.path(settings$host$outdir, run.id)
-   refcase      <- settings$model$binary
+   refcase      <- settings$model$binary # question1: refcase==casedir?
    bld          <- file.path(refcase,"bld")
    binary       <- file.path(bld,"cesm.exe")
-   indir        <- file.path(rundir,"input") ## input directory
+   indir        <- file.path(rundir,"data/shared") ## input directory
    default      <- settings$run$inputs$default$path ## reference inputs file structure
-   site_name    <- paste0(site.id %/% 1000000000, "-", site.id %% 1000000000)
+   site_name    <- paste0(site.id, "_c", run.id) ## change3
```
Comment (mdietze) [inline models/fates/R/write.configs.FATES.R:55]: I'm a bit concerned about the logic change here and on the casedir line. Could you explain what you're intending? Be aware that changes of internal logic can't change the nature of what PEcAn is passing into each function.

File: models/fates/R/write.configs.FATES.R [hunk 6]
```diff
@@ -116,59 +133,53 @@ write.config.FATES <- function(defaults, trait.values, settings, run.id){
...
-   if (!is.null(settings$host$postrun)) {
-     hostteardown <- paste(hostteardown, sep="\n", paste(settings$host$postrun, collapse="\n"))
-   }
+  # hostteardown <- ""
+  # if (!is.null(settings$model$postrun)) {
+  #   hostteardown <- paste(hostteardown, sep="\n", paste(settings$model$postrun, collapse="\n"))
+  # }
+  # if (!is.null(settings$host$postrun)) {
+  #   hostteardown <- paste(hostteardown, sep="\n", paste(settings$host$postrun, collapse="\n"))
+  # }
```
Comment (mdietze) [inline models/fates/R/write.configs.FATES.R:151]: For all these large chunks of code that you've commented out, are these things that are no longer used by CLM-FATES?

File: models/fates/R/write.configs.FATES.R [hunk 7]
```diff
@@ -202,143 +213,94 @@ write.config.FATES <- function(defaults, trait.values, settings, run.id){
...
-     photo_flag <- ncdf4::ncvar_get(fates.param.nc,varid="fates_c3psn", start = ipft, count = 1)
-     PEcAn.logger::logger.debug(paste0("Photosynthesis pathway flag value: ", photo_flag))
+     #photo_flag <- ncdf4::ncvar_get(fates.param.nc,varid="fates_c3psn", start = ipft, count = 1)
+     #PEcAn.logger::logger.debug(paste0("Photosynthesis pathway flag value: ", photo_flag))
      
      ## Loop over VARIABLES
-     for (v in seq_along(pft)) {
-       var <- names(pft)[v]
+     #for (v in seq_along(pft)) {
+       #var <- names(pft)[v]
```
Comment (mdietze) [inline models/fates/R/write.configs.FATES.R:275]: Similar to previous comment, what's the reason for commenting out the last ~50 lines?

File: models/fates/R/write.configs.FATES.R [hunk 8]
```diff
@@ -202,143 +213,94 @@ write.config.FATES <- function(defaults, trait.values, settings, run.id){
...
 
        ## THESE NEED SOME FOLLOW UP
        
        ### ----- Leaf physiological parameters
        # Vcmax
-       if(var == "Vcmax"){
-         ncdf4::ncvar_put(nc=fates.param.nc, varid='fates_vcmax25top', start = ipft, count = 1,
-                   vals=pft[v])  ## (umol CO2 m-2 s-1)
-       }
```
Comment (mdietze) [inline models/fates/R/write.configs.FATES.R:272]: have all of these variable names changed or are you pursuing a conceptually different approach for mapping PEcAn/BETY parameters to fates parameters?

File: models/fates/R/write.configs.FATES.R [hunk 9]
```diff
@@ -815,32 +384,16 @@ write.config.FATES <- function(defaults, trait.values, settings, run.id){
...
        
-     } ## end loop over VARIABLES
-   } ## end loop over PFTs
+     #} ## end loop over VARIABLES
+   #} ## end loop over PFTs
    #ncdf4::nc_close(param.nc)
-   ncdf4::nc_close(clm.param.nc)
-   ncdf4::nc_close(fates.param.nc)
+   #ncdf4::nc_close(clm.param.nc)
+   #ncdf4::nc_close(fates.param.nc)
```
Comment (mdietze) [inline models/fates/R/write.configs.FATES.R:393]: got to the end of this and do see where a new alternative way of parameter mapping ended up being implemented

File: modules/assim.sequential/DESCRIPTION [hunk 10]
```diff
@@ -32,7 +32,6 @@ Imports:
     stringr
 Suggests:
     corrplot,
-    exactextractr,
```
Comment (mdietze) [inline modules/assim.sequential/DESCRIPTION:35]: Why does this PR require changes to assim.sequential?

**Review by mdietze — CHANGES_REQUESTED** [review]
There's a lot that confuses me in this PR. First, it looks like this code is behind the mainline as it's deleted a lot of other people's changes. Second, it's deleted a ton of the FATES coupler without a clear substitution of an alternative structure for solving the same problem.

**Comment by Hhh-hyc** [comment]
Thanks for your detailed comment! @mdietze 

This PR is to show the steps in updating write.config file, and currently it's on the first stage to run the basic workflow of fates, where many fates coupler is uncommented for clarity and would be completely updated in the end. One problem is since my docker got stuck to open BETY webpage, temporarily some local address is used and name in the script is changed to run the fates model, just as you mentioned in the above comments. I would be grateful if you could have any idea in helping me solve the issue related to BETY.

As for the deleted contributions, I'm also confused cause I pushed my changes after having pulled the develop branch from upstream pecan. But I will check the version of my repository, and see if there are some problems to be solved.

**Pull request closed without merging**